Source/processcontainers/implementations/CRunImplementation/CRunImplementation.h

```diff
@@ -0,0 +1,86 @@
+#include "processcontainers/ProcessContainer.h"
+
+// TODO: Find more elegant alternative
+extern "C" {
+    #include <crun/error.h>
+    #include <crun/container.h>
+    #include <crun/utils.h>
+    #include <crun/status.h>
+}
+
+namespace WPEFramework {
+namespace ProcessContainers {
+
+    class CRunNetworkInterfaceIterator : public NetworkInterfaceIterator
+    {
+    public:
+        CRunNetworkInterfaceIterator();
+        ~CRunNetworkInterfaceIterator();
+
+        std::string Name() const override;
+        uint32_t NumIPs() const override;
+        std::string IP(uint32_t id) const override;
+    };
+
+    class CRunContainer : public IContainer 
+    {
+    public:
+        CRunContainer(string name, string path, string logPath);
```

Comment:
If it is reference counted, usggest making the Constructor private, so you can only get instances of this interface/Object through a Create method, this way the object could not be potentially placed on a stack

Reply:
also maybe use const string& as parameters ?